rclcpp/include/rclcpp/create_subscription.hpp

```diff
@@ -64,10 +74,46 @@ create_subscription(
   using rclcpp::node_interfaces::get_node_topics_interface;
   auto node_topics = get_node_topics_interface(std::forward<NodeT>(node));
 
+  std::shared_ptr<rclcpp::topic_statistics::SubscriptionTopicStatistics<CallbackMessageT>>
+  subscription_topic_stats = nullptr;
+
+  if (rclcpp::detail::resolve_enable_topic_statistics(
+      options,
+      *node_topics->get_node_base_interface()))
+  {
+    std::shared_ptr<Publisher<statistics_msgs::msg::MetricsMessage>> publisher =
+      create_publisher<statistics_msgs::msg::MetricsMessage>(
+      node,
+      options.topic_stats_options.publish_topic,
+      qos);
```

Comment:
qq: why does the topics statistics publisher use the same qos as the subscription being created?

I haven't checked the topics statistics demo, but it seems that if different entities are publishing statistics using different QoS, a variety of problems can happen (e.g.: pub/sub not matching).

Reply:
That's a good question. It's possible we could provide publishing QoS options as part of the topic statistics options. What do you think is the best path forward?

Reply:
>  It's possible we could provide publishing QoS options as part of the topic statistics options. What do you think is the best path forward?

I think that adding an extra option for that (like `options.topic_stats_options.publish_topic`) is a good idea.